lyse/__init__.py

```diff
@@ -422,6 +423,11 @@ def __init__(self, h5_path, run_paths, no_write=False):
             run_paths = run_paths['filepath']
         self.h5_path = h5_path
         self.no_write = no_write
+        try:
+            h5py.File(h5_path, 'r')
+        except:
+            h5py.File(h5_path,'w')
```

Comment:
Similar to previous comment, use of a context manager here would be preferable. Also, you might want to consider using the `'a'` filemode (so that the file doesn't get truncated should it happen to be created by an external process in between the two h5py calls executing - which could conceivably be a significant time should there be contention on the zlock/h5lock lock).

I think it might also be appropriate to wrap the creation call in a check of the `no_write` argument and raise the `FileNotFoundError` should the file not exist and the `no_write` argument be true?